src/components/PersonalSettings.vue

```diff
@@ -26,30 +19,50 @@
 				@update:checked="onCheckboxChanged($event, 'link_preview_enabled')">
 				{{ t('integration_github', 'Enable GitHub link previews') }}
 			</NcCheckboxRadioSwitch>
-			<div v-show="!showOAuth"
-				class="line">
-				<label for="github-token">
-					<KeyIcon :size="20" class="icon" />
-					{{ t('integration_github', 'Personal access token') }}
-				</label>
-				<input id="github-token"
-					v-model="state.token"
+			<NcNoteCard v-if="showOAuth && !connected" type="info">
+				{{ t('integration_github', 'You can manually provide a personal access token or connect via OAuth ') }}
+			</NcNoteCard>
+			<NcNoteCard v-if="!connected" type="info">
+				{{ t('integration_github', 'When you create a personal access token yourself, give it at least "read:user", "user:email" and "notifications" permissions.') }}
+				<a href="https://github.com/settings/tokens" target="_blank" class="external">
+					<span class="icon icon-external" />
+					{{ t('integration_github', 'GitHub personal access tokens') }}
+				</a>
+			</NcNoteCard>
+			<div v-if="!connected" class="line">
+				<NcTextField
+					id="github-token"
+					class="input"
+					:value.sync="state.token"
 					type="password"
-					:disabled="connected === true"
-					:placeholder="t('integration_github', 'GitHub personal access token')"
-					@keyup.enter="onConnectClick"
-					@focus="readonly = false">
+					:label="t('integration_github', 'Personal access token')"
+					placeholder="..."
+					:show-trailing-button="!!state.token"
+					@keyup.enter="connectWithToken"
+					@trailing-button-click="state.token = ''">
+					<KeyIcon />
+				</NcTextField>
+				<NcButton v-if="!connected"
+					:disabled="loading || state.token === ''"
+					:class="{ loading }"
+					@click="connectWithToken">
+					<template #icon>
+						<OpenInNewIcon :size="20" />
+					</template>
+					{{ t('integration_github', 'Connect to GitHub with a personal token') }}
+				</NcButton>
 			</div>
-			<NcButton v-if="!connected"
+			<NcButton v-if="showOAuth && !connected"
 				:disabled="loading === true || (!showOAuth && !state.token)"
```

Comment:
`loading === true || !!state.token"`

since false showOAuth would hide the element, and a populated token field should disable this btn

Reply:
Yup. Since there are 2 buttons now, the oauth one should be disabled if the token is defined. If that's what you mean, done.